pgmpy/base/DAG.py

```diff
@@ -1639,6 +1639,142 @@ def edge_strength(self, data, edges=None):
 
         return strengths
 
+    def _get_non_descendants(self, node):
+        descendants = set()
+        queue = list(self.succesors(node))
+        visited = set()
+
+        while queue:
+            current = queue.pop(0)
+            if current not in visited:
+                visited.add(current)
+                descendants.add(current)
+                queue.extend(self.successors(current))
+
+        all_nodes = set(self.nodes())
+        non_descendants = all_nodes - descendants - {node}
+        return list(non_descendants)
+
+    def count_lmc_violations(self, ci_test_func: Callable, significance_level=0.05):
```

Comment:
This is not really a property of DAG, so should be in the same file as the `permutation_test`.